src/com/gfredericks/test/chuck/generators.clj

```diff
@@ -252,3 +255,57 @@
   (gen/fmap (fn [ks]
               (select-keys m ks))
             (subsequence (keys m))))
+
+(def ^:private valid-offset-fns #{ct/millis ct/seconds ct/minutes ct/hours ct/days ct/months ct/years})
+
+(defn gen-datetime
+  "Generates datetime within given range and format.
+
+   base-datetime => By default it'll calculate the dates from year 2000.
+                    Generally this is a good idea instead of using (ct/now)
+                    since giving the same seed will generate the same output.
+                    If you would like to generate from a differnt base-datetime,
+                    Pass for example (ct/now) to use current time,
+                    Or pass a specific date-time (ct/date-time 2011 11 24)
+
+   offset-min & offset-max => The offset number range
+                              By default it is -1000 to 1000
+
+   offset-fns => List of functions which will be used with the given offset.
+                 It randomly picks one of the functions and
+                 applies the random offset with the given range.
+                 Check valid-offset-fns for possible values.
+                 By default its all the values of valid-offset-fns.
+
+   format-fn => Function to format the dt obj into.
+                By default it is ctc/to-string.
```

Comment:
another line of reasoning (for libraries in particular) is to hold off on extra features you're not sure you'll need, because it's much easier (w.r.t. backwards compatibility) to add them later if you change your mind, than vice versa.